modules/xmpp/JingleSessionPC.js

```diff
@@ -733,25 +756,21 @@ export default class JingleSessionPC extends JingleSession {
         // the assumption that candidates are spawned after the offer/answer
         // and XMPP preserves order).
         const workFunction = finishedCallback => {
-            if (!this._assertNotEnded('addIceCandidate')) {
-                finishedCallback();
-
-                return;
-            }
+            const allPromises = [];
 
             for (const iceCandidate of iceCandidates) {
-                this.peerconnection.addIceCandidate(iceCandidate)
-                    .then(
-                        () => logger.debug('addIceCandidate ok!'),
-                        err => logger.error('addIceCandidate failed!', err));
+                allPromises.push(
+                    this.peerconnection.addIceCandidate(iceCandidate)
+                        .then(
+                            () => logger.debug('addIceCandidate ok!'),
+                            err => logger.error('addIceCandidate failed!', err)));
             }
 
-            finishedCallback();
+            Promise.all(allPromises).then(finishedCallback);
```

Comment:
this is another change in behavior (maybe a bug fix?).  did you find an issue with this part? if not i'm wondering if we should just leave it for now.

Reply:
well yeah I can leave it, but then it will never be fixed and I think it is correct to wait for all promises to happen

Reply:
I can put into separate commit if that would be better, I can also remove the change if you insist

Reply:
just nervous about all the changes in this commit since the main one is potentially pretty big, thinking it would be nice to isolate.  these bugs (and i do agree, btw, that they are likely bugs) are either 1) not TOO drastic, since we haven't run into them or 2) done this way because of other weird lib-jitsi-meet behavior